### DISCUSSION TITLE
Should v3 Upgrader skip steps that don't apply to the current codebase?

### DISCUSSION THREAD

**Discussion by markjaquith** [livewire/livewire#5940] [3. Feature ideas]
The Livewire v3 upgrader is great, but there are a lot of steps. When running it through my codebase, it prompted me to run many steps that ended up having no effect.

Might it be better to have the replacement patterns do a proactive scan, and skip the steps that will end up having no effect, so fewer questions get asked of the user?

The scan wouldn't even have to be complete, as it could bail on the scan and prompt the user as soon as it found one replacement from the current step.

**Comment by joshhanley** [comment]
@PhiloNL what's your thoughts on this?

**Comment by PhiloNL** [comment]
I think it's a good suggestion. However, my concern is that most people will not bother to read the upgrade guide and assume they are done because the command says so; however, maybe the command didn't detect everything. If a step is not detected, a user is also unaware of why the upgrade does not seem to work, given they are unaware of the skipped step.

So in its current state, the command also acts as a “teacher” and prevents issues from being opened because people didn't read.

So, again, I appreciate the suggestion, but for now, I prefer to leave it as is.

**Reply by joshhanley** [reply]
Thanks!
